packages/lodestar/src/sync/utils/sync.ts

```diff
@@ -0,0 +1,83 @@
+import {IReputation, ReputationStore} from "../IReputation";
+import {BeaconBlock, BeaconBlockHeader, Checkpoint, Epoch, Slot} from "@chainsafe/eth2.0-types";
+import {IBeaconConfig} from "@chainsafe/eth2.0-config";
+import {hashTreeRoot} from "@chainsafe/ssz";
+import {blockToHeader} from "../../chain/stateTransition/util";
+import {IReqResp} from "../../network";
+
+export function isValidHeaderChain(config: IBeaconConfig, start: BeaconBlockHeader, blocks: BeaconBlock[]): boolean {
+  blocks = blocks.sort((b1, b2) => b1.slot - b2.slot);
+  let previousRoot = hashTreeRoot(start, config.types.BeaconBlockHeader);
+  for(const block of blocks) {
+    if(!previousRoot.equals(block.parentRoot)) {
+      return false;
+    }
+    previousRoot = hashTreeRoot(blockToHeader(config, block), config.types.BeaconBlockHeader);
+  }
+  return true;
+}
+
+export function getTargetEpoch(peers: IReputation[], currentCheckPoint: Checkpoint): Epoch {
+  const peersWithHigherFinalizedEpoch = peers.filter(peer => {
+    if(!peer.latestHello) {
+      return false;
+    }
+    if(peer.latestHello.finalizedEpoch > currentCheckPoint.epoch) {
+      return true;
+    }
+  });
+  if(peersWithHigherFinalizedEpoch.length > 0) {
+    return currentCheckPoint.epoch + 1;
```

Comment:
I don't understand the + 1 logic.

I think we need to work with a Checkpoint, ie not throw away the root, because the final part of the fast sync strategy is to check the root against the last downloaded block to ensure we've downloaded the chain we think we've downloaded.

Reply:
there is no method to fetch non latest finalized checkpoint from peers. I'll extract +1 to constant, basically if you put +2 here it will sync 2 epoch at a time. We can only sync next epoch if there is peer with higher finalized epoch